src/common/VirtualTerminal.cpp

```diff
@@ -187,6 +187,8 @@ namespace SDDM {
                 qWarning("Couldn't finalize jump to VT %d: %s", vt, strerror(errno));
 
             close(activeVtFd);
+            if (fd != activeVtFd)
```

Comment:
This might also be:
  if (vtFd != -1) close(vtFd);

It seems slightly more obvious this way, but no strong opinions.

Would be good to hold these into a unique_fd class like this one:
  https://cs.android.com/android/platform/superproject/+/master:system/libbase/include/android-base/unique_fd.h

I don't know whether Qt or KDE Frameworks has anything similar.

Reply:
`QFile` would be ideal, but it doesn't support passing `O_NOCTTY`...

Reply:
Changed